src/tsdb/prometheus.act

```diff
@@ -0,0 +1,147 @@
+#
+
+import acton.rts
+import file
+import net
+import time
+import snappy
+import logging
+import http
+from tsdb.common import Metric
+
+#class Metric(object):
+#    @property
+#    name: str
+#    @property
+#    tags: dict[str, str]
+#    @property
+#    value: float
+#    @property
+#    timestamp: int
+#
+#    def __init__(self, name: str, tags: dict[str, str], value: float, timestamp: int):
+#        self.name = name
+#        self.tags = tags
+#        self.value = value
+#        self.timestamp = timestamp
+#
+#    def __str__(self):
+#        # XXX including str(self.timestamp) in this repr triggers a segfault for reasons unknown
+#        return "<tsdb.Metric" + " " + str(self.name) + " " + str(self.value) + " " + str(self.tags) + " " + str(self.timestamp) + ">"
+#
+#    def __repr__(self):
+#        return self.__str__()
+
+#actor Client(auth: net.TCPConnectCap, address: str, port: int, on_connect: action(Client) -> None, on_error: action(Client, str) -> None):
+
+def pack(time_series: list[(list[(str, str)], list[(float, int)])]) -> bytes:
+    #return "".encode()
+    NotImplemented
+
+actor Client(auth: net.TCPConnectCap, address: str, port: int, api_endpoint: str, on_connect: action(Client) -> None, on_error: action(Client, str) -> None, log_handler: ?logging.Handler):
+
+    logh = logging.Handler("tsdb.prometheus.Client")
+    if log_handler is not None:
+        logh.set_handler(log_handler)
+    log = logging.Logger(logh)
+
+    def _on_connect(client: http.Client):
+        log.info("TSDB HTTP client connected!", None)
+        await async on_connect(self)
+
+    def _on_error(client: http.Client, error: str):
+        log.error("TSDB HTTP client connection failed!", None)
+        await async on_error(self, error)
+
+    log.info("Launching TSDB HTTP client", None)
+    http_client = http.Client(auth, address, _on_connect, _on_error, "http", port)
+
+    def quick_init(on_response: action(Client, bool) -> None):
+        def _on_response(client: http.Client, response: http.Response):
+            log.debug("HTTP response status: " + str(response.status), None)
+            log.trace(response.body.decode(), None)
+
+            def _post_ready(timeout):
+                json_dict = { "name": "default" }
+                #http_client.post("/api/v1/services/prometheusdb/namespace/ready", json.encode(json_dict).encode(), lambda c, r: _on_ready_response(c, r, timeout))
+
+            def _on_ready_response(client: http.Client, response: http.Response, timeout: int):
+                log.trace("Namespace ready request:" + response.body.decode(), None)
```

Comment:
Do not str format data into log message, send as second arg, see https://github.com/telemetrify-collector/telemetrify/pull/63/files#r1524463591